alarm/PolicyManager2.js

```diff
@@ -195,8 +198,17 @@ class PolicyManager2 {
 
   checkAndSave(policy, callback) {
     callback = callback || function() {}
-
-    this.savePolicy(policy, callback);
+    async(()=>{
+      //FIXME: data inconsistence risk for multi-processes or multi-threads
+      let policies = await(this.getSimilarPolicies(policy))
+      if (policies && policies.length > 0) {
+        log.info("policy with type:" + policy.type + ",target:" + policy.target + " already existed")
+        policy.pid = policies[0].pid
+        callback(null, policy.pid)
```

Comment:
this might introduce a issue that multiple alarms are bound to same policy, need to think about how to handle this use case.

Reply:
What about let client create and save duplicate policies each time, but will check whether there are similar policies with same type and target when doing enforcement? Also, when client need delete a policy, need delete all similar policies.

Reply:
this will make it even more complex, need a simpler solution.